src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -103,11 +98,7 @@ public T set(T value, int index) {
 
     @Override
     public T remove(int index) {
-        if (index < size && index >= 0) {
-            return unlink(getNodeByIndex(index));
-        } else {
-            throw new IndexOutOfBoundsException("Such index does not exist!");
-        }
+        return indexExists(index) ? unlink(getNodeByIndex(index)) : null;
```

Comment:
indexExists(index);
return unlink(getNodeByIndex(index));

same here, if your index is invalid the exception will be thrown before the return statement